firebase-firestore/api.txt

```diff
@@ -19,13 +19,39 @@ package com.google.firebase {
 
 package com.google.firebase.firestore {
 
+  public abstract class AggregateField {
+    method @NonNull public static com.google.firebase.firestore.AggregateField.AverageAggregateField average(@NonNull String);
+    method @NonNull public static com.google.firebase.firestore.AggregateField.AverageAggregateField average(@NonNull com.google.firebase.firestore.FieldPath);
+    method @NonNull public static com.google.firebase.firestore.AggregateField.CountAggregateField count();
+    method @NonNull @RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY) public String getAlias();
+    method @NonNull @RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY) public String getFieldPath();
+    method @NonNull @RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY) public String getOperator();
+    method @NonNull public static com.google.firebase.firestore.AggregateField.SumAggregateField sum(@NonNull String);
+    method @NonNull public static com.google.firebase.firestore.AggregateField.SumAggregateField sum(@NonNull com.google.firebase.firestore.FieldPath);
+  }
+
+  public static class AggregateField.AverageAggregateField extends com.google.firebase.firestore.AggregateField {
+  }
+
+  public static class AggregateField.CountAggregateField extends com.google.firebase.firestore.AggregateField {
+  }
+
+  public static class AggregateField.SumAggregateField extends com.google.firebase.firestore.AggregateField {
+  }
+
   public class AggregateQuery {
     method @NonNull public com.google.android.gms.tasks.Task<com.google.firebase.firestore.AggregateQuerySnapshot> get(@NonNull com.google.firebase.firestore.AggregateSource);
+    method @NonNull public java.util.List<com.google.firebase.firestore.AggregateField> getAggregateFields();
     method @NonNull public com.google.firebase.firestore.Query getQuery();
   }
 
   public class AggregateQuerySnapshot {
+    method @Nullable public Object get(@NonNull com.google.firebase.firestore.AggregateField);
+    method @RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY) public long get(@NonNull com.google.firebase.firestore.AggregateField.CountAggregateField);
+    method @Nullable @RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY) public Double get(@NonNull com.google.firebase.firestore.AggregateField.AverageAggregateField);
```

Comment:
Should these two API members still have the annotation `@RestrictTo(androidx.annotation.RestrictTo.Scope.LIBRARY)`?

Reply:
Can you tell me which of these APIs need to be restricted? I checked the API proposal and I am confused :(